src/plugins/profiling/INTERNALS.md

```diff
@@ -0,0 +1,105 @@
+This plugin is divided into two components: the UI and the server.
+
+The UI is responsible for rendering the charts and flamegraphs. It will make
+API calls to the server component depending on the user interactions with the
+UI. You will find most of the source code in the `public` directory.
+
+The server is responsible for retrieving data from the necessary sources and
+sending it in the expected format to the UI in response to the UI's API calls.
+You will find most of the source code in the `server` directory.
+
+## Server API
+
+Depending on how the plugin is configured, there are two different sets of API
+calls.
+
+If the server uses local fixtures as a data source, then the following API is
+served by the server and called by the UI:
+
+* /api/prodfiler/v1/topn/containers
+* /api/prodfiler/v1/topn/deployments
+* /api/prodfiler/v1/topn/hosts
+* /api/prodfiler/v1/topn/threads
+* /api/prodfiler/v1/topn/traces
+* /api/prodfiler/v1/flamechart/elastic
+* /api/prodfiler/v1/flamechart/pixi
+
+If the server uses an Elasticsearch cluster as a data source, then the following
+API is served by the server and called by the UI:
+
+* /api/prodfiler/v2/topn/containers
+* /api/prodfiler/v2/topn/deployments
+* /api/prodfiler/v2/topn/hosts
+* /api/prodfiler/v2/topn/threads
+* /api/prodfiler/v2/topn/traces
+* /api/prodfiler/v2/flamechart/elastic
+* /api/prodfiler/v2/flamechart/pixi
+
+By default, the plugin is configured to use the second API set. See README.md to
+configure the plugin to use the first API set (aka local fixtures as a data
+source).
+
+Both API sets are expected to return the same response format.
+
+The design to have separate API sets for local vs Elasticsearch was partly
+because the UI and server components were originally developed separately and
+later merged. However, it also allows the server methods to have a single
+responsibility and makes it easier to test and verify both the server returns
```

Comment:
Not sure, maybe there is a dot missing before `the server ...` ?

Reply:
I think I had one sentence in my head and wrote another one, so the sentence was confusing. I hope it's clearer now.